src/main/scala/io/qbeast/spark/delta/writer/BlockWriter.scala

```diff
@@ -140,16 +140,20 @@ case class BlockWriter(
               |size=${file.size},
               |modificationTime=${file.modificationTime},
               |revision=${file.revisionId}""".stripMargin.replaceAll("\n", " "))
-      val addFile = IndexFiles.toAddFile()(file)
+      val addFile = IndexFiles.toAddFile(dataChange = true)(file)
```

Comment:
Could this also be executed during an optimization step, in which case data change should be false? I'm a bit unclear on the criteria to determine whether data change should be true or false in this code.

Reply:
`BlockWriter` for some reason is not currently used anywhere. As files contain more than one block, all writes are handles through `IndexFileWriter.`

Reply:
OK, in this case is the correct course of action to deprecate `BlockWriter`? If you confirm, then let's please create an issue to remove it separately.

Otherwise, I'm unsure about the correct way to set the flag here. If we are to deprecate this class to then remove, then we don't need to worry about maintaining it.

Reply:
I opened issue #314 for this.

Reply:
As per @Jiaweihu08 offline, it is making the value of the flag explicit.

Reply:
As per @Jiaweihu08 offline, it is making the value of the flag explicit.